Completely remove FutureMessage from RPC TorchScript implementations

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50236 Apply clang-format to rpc cpp files
* #50029 Completely remove FutureMessage type
* #50028 Completely Remove FutureMessage from RPC agents
* #50027 Completely Remove FutureMessage from RPC cpp tests
* #50026 Remove FutureMessage from RPC request callback logic
* #50025 Completely remove FutureMessage from FaultyProcessGroupAgent
* #50024 Remove FutureMessage from sender TensorPipeAgent
* #50023 Remove FutureMessage from sender ProcessGroupAgent
* #50020 Completely remove FutureMessage from distributed autograd
* **#50005 Completely remove FutureMessage from RPC TorchScript implementations**
* #50004 Completely remove FutureMessage from RRef Implementations
* #49995 Replace FutureMessage with ivalue::Future in RpcAgent retry logic
* #49960 Replace FutureMessage with ivalue::Future in RRefContext
* #49927 Replace FutureMessage with ivalue::Future in distributed/autograd/utils.*
* #49906 Let RpcAgent::send() return JitFuture

Differential Revision: [D25750663](https://our.internmc.facebook.com/intern/diff/D25750663)

mrshenli committed:
Completely remove FutureMessage from RPC TorchScript implementations

[ghstack-poisoned]

mrshenli committed:
Update on "Completely remove FutureMessage from RPC TorchScript implementations"


Differential Revision: [D25750663](https://our.internmc.facebook.com/intern/diff/D25750663)

[ghstack-poisoned]

mrshenli committed:
Update on "Completely remove FutureMessage from RPC TorchScript implementations"


Differential Revision: [D25750663](https://our.internmc.facebook.com/intern/diff/D25750663)

[ghstack-poisoned]

lw commented on torch/csrc/distributed/rpc/torchscript_functions.cpp:
```diff
@@ -64,14 +63,14 @@ c10::intrusive_ptr<c10::ivalue::Future> rpcTorchscript(
-    auto futMessage = wp.lock();
-    if (futMessage->hasError()) {
-      c10::ivalue::Future::FutureError jitFutErr(futMessage->error()->what());
-      futPtr->setError(std::make_exception_ptr(jitFutErr));
+  std::weak_ptr<JitFuture> wp = jitFuture;
```
Nit: this code is mixing the `c10::ivalue::Future` and the `JitFuture` names, although the latter is just an alias for the former. An unaware reader might get confused by this. Are there plans to later on replace all the JitFuture occurrences with ivalue::Future?

mrshenli replied:
Yep, will add a few followup PRs: 1) replace utils/future.h with ivalue::Future in RRef, 2) remove utils/future.h, 3) replace `shared_ptr<JitFuture>` with `intrusive_ptr<JitFuture>`, 4) rename `JitFuture` occurrences to `ivalue::Future`. 

But let me make this PR consistent first.

mrshenli committed:
Update on "Completely remove FutureMessage from RPC TorchScript implementations"


Differential Revision: [D25750663](https://our.internmc.facebook.com/intern/diff/D25750663)

[ghstack-poisoned]

mrshenli committed:
Update on "Completely remove FutureMessage from RPC TorchScript implementations"


Differential Revision: [D25750663](https://our.internmc.facebook.com/intern/diff/D25750663)

[ghstack-poisoned]
